compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -349,7 +348,7 @@ trait QuotesAndSplices { self: Typer =>
     UnApply(
       fun = ref(defn.InternalQuotedMatcher_unapply.termRef).appliedToTypeTrees(typeBindingsTuple :: TypeTree(patType) :: Nil),
       implicits =
-        ref(defn.InternalQuoted_exprQuote.termRef).appliedToType(shape.tpe).appliedTo(shape) ::
+          ref(defn.InternalQuoted_exprQuote.termRef).appliedToType(defn.AnyType).appliedTo(shape).select(nme.apply).appliedTo(qctx) ::
```

Comment:
Why `defn.AnyType` instead of `shape.tpe`?

Reply:
`shape.tpe` was not avoiding some types, but as the parameter is of type `Expr[_]` it could be solved in a simpler and more efficient way. Nothing else depends on this type.